app/webpacker/components/Panel/pages/ApprovePictures/index.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useState } from 'react';
+import {
+  Button,
+  CardGroup,
+  Header,
+  List,
+  Segment,
+} from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import I18n from '../../../../lib/i18n';
+import AvatarCard from './AvatarCard';
+import updateAvatars from './api/updateAvatars';
+import WCAQueryClientProvider from '../../../../lib/providers/WCAQueryClientProvider';
+
+const transformAvatarsForRequest = (approvedAvatars, rejectedAvatars) => {
+  const avatars = {};
+
+  approvedAvatars.forEach(({ id }) => {
+    avatars[id] = { action: 'approve' };
+  });
+
+  rejectedAvatars.forEach(({ id, rejectionReason }) => {
+    avatars[id] = { action: 'reject', rejection_reason: rejectionReason };
+  });
+
+  return { avatars };
+};
+
+export default function Wrapper({ pendingUsers }) {
+  return (
+    <WCAQueryClientProvider>
+      <ApprovePictures pendingUsers={pendingUsers} />
+    </WCAQueryClientProvider>
+  );
+}
+
+function ApprovePictures({ pendingUsers }) {
+  const [approvedAvatars, setApprovedAvatarIds] = useState([]);
+  const [rejectedAvatars, setRejectedAvatarIds] = useState([]);
+  const [deferredAvatars, setDeferredAvatarIds] = useState([]);
```

Comment:
I assume each avatar should be in at most one of these lists? A reducer could enforce that, but it might be over-kill in this case.

Also: inconsistency in the names - `...Avatars` vs `AvatarIds`.

Also also: the third state is unused?

Reply:
I will remove all these state because we don't actually need batch processing